src/coreclr/jit/emit.cpp

```diff
@@ -5179,16 +5183,35 @@ unsigned emitter::getLoopSize(insGroup* igLoopHeader, unsigned maxLoopSize DEBUG
             else
 #endif
             {
+                JITDUMP(" but ends with align instruction, taking off %u bytes.",
+                        emitComp->opts.compJitAlignPaddingLimit);
                 // The current loop size should exclude the align instruction size reserved for next loop.
                 loopSize -= emitComp->opts.compJitAlignPaddingLimit;
             }
         }
+        else if (igInLoop->igFlags & IGF_REMOVED_ALIGN)
+        {
+            assert("!Failed to remove 15 bytes");
```

Comment:
Is this really always 15 bytes?

Reply:
It is `compJitAlignPaddingLimit` but this code is just for testing purpose. I will remove it after I see the CI result.

Reply:
OK.
I'm interested to see if you can manage to remove instructions cleanly. I've been trying to do that with unneeded jmps but I've found that doing it breaks exception handling completely.

Reply:
```suggestion
            assert(!"Failed to remove 15 bytes");
```

Reply:
good catch. Thanks! :)